docs/upgrade/0.3.x-to-0.3.5.rst

```diff
@@ -0,0 +1,119 @@
+a reboot would interrupt a running upgrade for cron-apt. The new logic
+ensures that the reboot only takes place after cron-apt exits.
+
+Blacklist additional kernel modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Newer versions of the Intel NUCs come with wifi and bluetooth hardware,
+so the playbooks now disable the ``btusb`` and ``iwlmvm`` kernel modules,
+in addition to ``bluetooth`` and ``iwlwifi``, which are still blacklisted.
+
+
+Prerequisites
+-------------
+
+-  An Admin networked Tails workstation with persistence enabled and an
+   Admin password set during boot.
+   
+-  A running SecureDrop instance of 0.3.x, where x is less than 5.
+
+.. note:: If your organization uses an internal SMTP relay, you may
+          wish to continue using fingerprint-style verification of the
+          TLS certificate. You can therefore skip #3 below.
+
+Upgrade Procedure
+-----------------
+
+The upgrade procedure can be performed entirely from the Admin
+Workstation. Start by booting the Admin Tails USB on the Admin 
+Workstation. Make sure you set an *Adminstrator Password* in the *Tails Greeter*.
+
+#. Open a terminal (it is an icon on the top of the screen that looks
+   like a little black TV).
+
+#. Change into the SecureDrop repo directory ::
+
+     cd /home/amnesia/Persistent/securedrop/install_files/ansible-base
+
+#. Open ``prod-specific.yml`` in your preferred text editor.
+   Delete the line that starts with ``smtp_relay_fingerprint`` and save the file.
+   If you are using your own or your organization's SMTP relay (not Google's),
+   you may want want to continue using fingerprint verification.
+   If you are unsure, contact us and we will help you make a determination.
+
+#. Stash your updated local changes ::
+
+     git stash save "site specific configs"
+
+#. Pull the latest code ::
+
+     git pull
+
+#. Verify the signed git tag for the latest stable release ::
+
+     git tag -v 0.3.5
+
+#. Checkout the latest release ::
+
+     git checkout 0.3.5
+
+#. Restore your instance-specific settings ::
+
+     git checkout stash -- prod-specific.yml inventory
+     git reset
+
+   .. note:: SecureDrop upgrades typically recommend using the simpler
+             ``git stash pop`` command to restore instance-specific settings.
+             Since 0.3.5 makes changes to the ``prod-specific.yml`` file,
+             more verbose git commands are necessary to avoid a merge conflict.
+
+#. Make sure you have Ansible installed. If running ``which ansible`` returns ``ansible not found``, you should ::
+
+    sudo apt-get update
+    sudo apt-get install ansible
+
+#. Run the Ansible playbooks (substitute the admin account on the servers for ``<user>``) ::
+
+    ansible-playbook -i inventory -s -u <user> securedrop-prod.yml
+
+During the playbook run, the Postfix TLS verification settings will be updated
+on the Monitor Server, the AppArmor profiles will be updated on the Application Server,
```

Comment:
Nit: the AppArmor profiles are updated automatically with version 0.3.5 of the `securedrop-app-code` package, which happens independently of the Ansible playbook run for 0.3.5.